mmdet/models/detectors/base.py

```diff
@@ -198,6 +198,13 @@ def _parse_losses(self, losses):
         loss = sum(_value for _key, _value in log_vars.items()
                    if 'loss' in _key)
 
+        # If the loss_vars has different length, GPUs will wait infinitely
+        if dist.is_available() and dist.is_initialized():
+            log_var_length = torch.tensor(len(log_vars), device=loss.device)
+            dist.all_reduce(log_var_length)
+            assert log_var_length == len(log_vars) * dist.get_world_size(), \
```

Comment:
Once an error occurs, print out all the keys in each process to facilitate troubleshooting.

Reply:
does mmdet provide any thread-safe print utilities?

Reply:
No need to think about it, only need to be printed after asset, and keys and rank need to be printed at the same time.

Reply:
Also, how can other GPUs know that this GPU raises error?

Reply:
It seems that if we want to inform all GPUs about the assertion error, we have to do one more communication among GPUs. If this overhead is Okay, I will add it to this PR.

Reply:
There is no need to distinguish which GPU is wrong, just print all the keys, the user can check the output to determine which key is wrong

`assert Flase, f'{...get_rank(), log_val.keys(), len(log_val.keys())...}'`

Reply:
Yes, you are right. At least two GPUs will raise exception, so users can compare the error messages.

Reply:
Yes